### DISCUSSION TITLE
Make it possible to dynamically register resources based on user

### DISCUSSION THREAD

**Discussion by enso-san** [filamentphp/filament#4173] [Ideas]
Right now its not possible to register different resources based on specific user roles.
The user data is only available when the StartSession middleware was executed, however registration of filament resources is only possible early from within a service provider which runs before.
Trying to use Filament::registerResource() at later times, will result in "route not defined" errors.

**Comment by danharrin** [comment]
Opened an issue - #4177

**Comment by danharrin** [comment]
Completed now, it will be available in v3 :)
